pages/login/index.js

```diff
@@ -38,84 +67,127 @@ const Index = () => {
+                      fill="currentFill"
+                    />
+                  </svg>
+                </div>
+              )}
+              Login
+            </button>
+          </div>
+          <div
+            className="mb-[1rem] flex w-[27.4rem] flex-col items-center rounded-[1.5rem]
+            border border-[#50545A] px-4 py-[1.1rem]"
+          >
+            <button
+              onClick={discordAuth}
+              className="mx-[1.1rem] h-[5rem] w-[25.2rem] w-full rounded-[1rem] bg-[#5865F2] text-[1.8rem] text-white"
+            >
+              Login with Discord
+            </button>
+          </div>
+          <div className="mb-[1rem] text-[1.6rem]">or</div>
           <button
-            onClick={discordAuth}
-            className="mx-[1.1rem] h-[5rem] w-[25.2rem] rounded-[1rem] bg-[#5865F2] text-[1.8rem] text-white"
+            onClick={signupInstead}
+            className="mx-[1.1rem] h-[5rem] w-[25.2rem] rounded-[1rem] border bg-black text-[1.8rem] text-white"
           >
-            Login with Discord
+            Sign Up
           </button>
         </div>
-        <div className="mb-[1rem] text-[1.6rem]">or</div>
-        <button
-          onClick={signupInstead}
-          className="mx-[1.1rem] h-[5rem] w-[25.2rem] rounded-[1rem] border bg-black text-[1.8rem] text-white"
-        >
-          Sign Up
-        </button>
+        <footer className="h-84 fixed bottom-0 flex hidden   w-full bg-gray-900 md:block">
+          <div className="h-full w-full ">
+            <Image
+              src="/images/gifs/moon-holdings-banner-wide.gif"
+              alt=""
+              height={0}
+              width={0}
+              className=" h-full w-full "
+            />
+          </div>
+        </footer>
       </div>
-
-      <div className="flex justify-center">
-        <Image
-          src="/images/gifs/moon-holdings-banner-wide.gif"
-          alt=""
-          className="z-{25} fixed bottom-0 hidden w-full max-w-[144rem] md:-bottom-[5rem] md:block md:w-full lg:-bottom-[8rem]"
-        />
-        <Image
-          src="/images/gifs/moon-holdings-banner-cropped.gif"
-          alt=""
-          className="z-{25} fixed bottom-0 w-full max-w-[144rem] md:hidden"
-        />
-      </div>
-    </div>
+      <BannerModal
+        showModal={showModal}
+        message={message}
+        hasError={error}
+        closeModal={() => {
+          setModal('', false, false)
+        }}
+      />
+    </>
   )
 }
 
-export default Index
+export default Login
+
+export async function getServerSideProps(context) {
+  const session = await getSession(context)
+
+  //if user is logged in
+  if (session) {
+    return {
+      redirect: {
+        destination: '/',
+        permanent: false,
+      },
+    }
+  } else {
+    return { props: {} }
+  }
+}
```

Comment:
I think we can create a reusable higher order component for this since it is being reused multiple times in different pages.. Something like `withAuth(Login)` hoc